tests/lib/ble_gq/src/unity_test.c

```diff
@@ -0,0 +1,309 @@
+	__cmock_sd_ble_gattc_characteristics_discover_ExpectAndReturn(
+		conn_handle, &req.gattc_char_disc, NRF_ERROR_INVALID_ADDR);
+
+	ret = ble_gq_item_add(&ble_gq, &req, 0);
+	TEST_ASSERT_EQUAL(0, ret);
+	TEST_ASSERT_EQUAL(conn_handle, glob_conn_handle);
+	TEST_ASSERT_EQUAL(NRF_ERROR_INVALID_ADDR, glob_error);
+}
+
+void test_ble_gq_item_add_req_desc_discovery(void)
+{
+	uint16_t conn_handle = 0;
+	int ret;
+	struct ble_gq_req req = {
+			.type = BLE_GQ_REQ_DESC_DISCOVERY,
+			.error_handler = {
+					.cb = ble_gq_error_handler,
+					.ctx = NULL
+			},
+			.gattc_desc_disc = {
+					.start_handle = 0,
+					.end_handle = 0
+			}
+	};
+	ble_gq.conn_handles[0] = 0;
+
+	__cmock_sd_ble_gattc_descriptors_discover_ExpectAndReturn(
+		conn_handle, &req.gattc_desc_disc, NRF_ERROR_BUSY);
+	__cmock_sd_ble_gattc_descriptors_discover_ExpectAndReturn(
+		conn_handle, &req.gattc_desc_disc, BLE_ERROR_INVALID_CONN_HANDLE);
+
+	ret = ble_gq_item_add(&ble_gq, &req, 0);
+	TEST_ASSERT_EQUAL(0, ret);
+	TEST_ASSERT_EQUAL(conn_handle, glob_conn_handle);
+	TEST_ASSERT_EQUAL(BLE_ERROR_INVALID_CONN_HANDLE, glob_error);
+}
+
+void test_ble_gq_item_add_req_gatts_hvx(void)
+{
+	uint8_t data[] = { 0x01, 0x02, 0x03 };
+	uint16_t len = sizeof(data);
+	uint16_t conn_handle = 0;
+	int ret;
+	struct ble_gq_req req = {
+		.type = BLE_GQ_REQ_GATTS_HVX,
+		.error_handler = {
+			.cb = ble_gq_error_handler,
+			.ctx = NULL
+		},
+		.gatts_hvx = {
+			.type = BLE_GATT_HVX_NOTIFICATION,
+			.handle = 0,
+			.offset = 0,
+			.p_data = data,
+			.p_len = &len
+		}
+	};
+	ble_gq.conn_handles[0] = 0;
+
+	__cmock_sd_ble_gatts_hvx_ExpectAndReturn(conn_handle, &req.gatts_hvx, NRF_ERROR_BUSY);
+	__cmock_sd_ble_gatts_hvx_IgnoreAndReturn(NRF_SUCCESS);
+	ret = ble_gq_item_add(&ble_gq, &req, 0);
+	TEST_ASSERT_EQUAL(0, ret);
+
+	__cmock_sd_ble_gatts_hvx_ExpectAndReturn(conn_handle, &req.gatts_hvx, NRF_ERROR_BUSY);
+	__cmock_sd_ble_gatts_hvx_IgnoreAndReturn(NRF_ERROR_INVALID_ADDR);
+
+	ret = ble_gq_item_add(&ble_gq, &req, 0);
+	TEST_ASSERT_EQUAL(0, ret);
+	TEST_ASSERT_EQUAL(conn_handle, glob_conn_handle);
+	TEST_ASSERT_EQUAL(NRF_ERROR_INVALID_ADDR, glob_error);
+
+	req.gatts_hvx.p_len = NULL;
+	ret = ble_gq_item_add(&ble_gq, &req, 0);
+	TEST_ASSERT_EQUAL(0, ret);
+	TEST_ASSERT_EQUAL(NRF_ERROR_INVALID_PARAM, glob_error);
+}
+
+void test_ble_gq_conn_handle_register_efault(void)
+{
+	int ret = ble_gq_conn_handle_register(NULL, 0);
+
+	TEST_ASSERT_EQUAL(-EFAULT, ret);
+}
+
+void test_ble_gq_conn_handle_register_enomem(void)
+{
+	int ret;
+
+	ret = ble_gq_conn_handle_register(&ble_gq, 0);
+	TEST_ASSERT_EQUAL(0, ret);
+
+	for (int i = 0; i < ble_gq.max_conns; i++) {
+		ble_gq.conn_handles[i] = i + 1;
+	}
+	ret = ble_gq_conn_handle_register(&ble_gq, 3);
+	TEST_ASSERT_EQUAL(-ENOMEM, ret);
+}
+
+void test_ble_gq_on_ble_evt(void)
+{
+	ble_evt_t ble_evt = {0};
+	uint16_t conn_handle = 0x0C4;
+
+	ble_evt.header.evt_id = BLE_GAP_EVT_DISCONNECTED;
+	ble_evt.evt.gap_evt.conn_handle = 0x0C4;
```

Comment:
Nit: I would consider defining `0x0C4` somewhere and then reusing the definition in all the different occurrences, rather than hardcoding the value.